src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/PackageRestore/PackageRestoreDataSource.cs

```diff
@@ -206,5 +291,160 @@ public Task UnloadAsync()
 
```

Comment:
just for the safeness , inside UnloadAsync, 

_whenNominatedTask?TrySetCanceled();

If anyone waits it when NuGet restore is disabled, abort the waiting task.